src/main/java/org/killbill/billing/plugin/currency/dao/CurrencyDao.java

```diff
@@ -73,22 +73,27 @@ public void run(final Configuration configuration) {
                                                      toLocalDateTime(utcNow),
                                                      toLocalDateTime(utcNow),
                                                      kbTenantId == null ? null : kbTenantId.toString())
+                                             .onDuplicateKeyUpdate()
```

Comment:
Interesting approach. This doesn't match the PR description 😄 

This only works if the unique index exists. Update `ddl.sql` to add a unique index on `base_currency, conversion_date, kb_tenant_id` for the table `currency_updates`. Write a new unit test in `TestStaticCurrencyPluginApi` to verify the scenario.